api/runner/runner.go

```diff
@@ -29,6 +29,7 @@ type Runner struct {
 	flog         FuncLogger
 	availableMem int64
 	usedMem      int64
+	Tasks        chan task.Request
```

Comment:
> Shouldn't code be closed for modification and open for extension? Or putting differently, if you are exposing an internal part like this, you are also saying for future customizers of IronFunction that a) Tasks can be replaced at any time (which his not true); b) Tasks do not need any protection when being changed (channel itself is safe, but if you can just replace the channel with something else, then you could have a race condition here).

Well, the task channel was exposed either way. I still think that it could be created and kept inside the `runner` and we could easily test its functionality during runner tests.

I still don't understand why do you think keeping the task channel inside the runner prevent us from testing it correctly.

> That's not true - or I might have read your changes incorrectly. With the external channel, there were mocks that pretended that tasks had been consumed. In this PR, you are just running these tasks in the Docker again, no?

I afraid you are correct. Reviewing my changes I confirm I forgot to reproduce that behavior of mocking in most of the cases but one particular case.

But still think even with these changes, that mocking can be done with the tasks embedded in the runner.

> I clearly failed to convey my concern, and I apologize. You are just stating that your change is better (embedding), but I did not see anywhere why before was worse. You essentially took a channel that was meant to link together two different components, and put in one of them - in practice defeating the meaning of it being external: show the connection of two independent executing processes.
> Could you please elaborate on this? It seems something important. Perhaps some example?

IMO the task channel belongs to the runner and all components that were dependant to the task channel were also dependant to the `runner`. So why really keep both separated?

I agree we can improve it more by, for example, not embed the `runner` inside the `server` and just pass a channel for communication. Same for other comments that would decentralize things and keep it more simple.

Do you think we should proceed that direction?